File: VhallAndroidSDKLiveDemo/app/src/main/java/com/vhall/live/LoginActivity.java
package com.vhall.live;

import android.app.Activity;
import android.app.AlertDialog;
import android.app.ProgressDialog;
import android.content.DialogInterface;
import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.MotionEvent;
import android.view.View;

import android.widget.EditText;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.TextView;
import android.widget.Toast;

import com.vhall.business.VhallSDK;
import com.vhall.business.data.ILoginRequest;
import com.vhall.business.data.UserInfo;
import com.vhall.business.data.source.UserInfoDataSource;

/**
 * 登录界面
 */
public class LoginActivity extends Activity implements RadioGroup.OnCheckedChangeListener {

    private static final String TAG = "LoginActivity";
    private EditText mTextInputUsername;
    private EditText mTextInputPassword;
    public AlertDialog alertDialog;
    private TextView mVerTextView;
    private RadioButton account_login, id_login;
    private RadioGroup login_type_container;
    private EditText head_net_url, third_id;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.login_activity);
        mTextInputUsername = this.findViewById(R.id.text_input_username);
        mTextInputPassword = this.findViewById(R.id.text_input_password);
        mVerTextView = this.findViewById(R.id.label_ver);
        account_login = this.findViewById(R.id.account_login);
        id_login = this.findViewById(R.id.id_login);
        head_net_url = this.findViewById(R.id.head_net_url);
        third_id = this.findViewById(R.id.third_id);
        login_type_container = this.findViewById(R.id.login_type_container);
        mVerTextView.setText(VhallSDK.getVersion());

        login_type_container.setOnCheckedChangeListener(this::onCheckedChanged);
        mWaitingDialog = new ProgressDialog(this);
        mWaitingDialog.setTitle("提示");
        mWaitingDialog.setMessage("请稍后...");
    }

    @Override
    public void onCheckedChanged(RadioGroup group, int checkedId) {
        if (checkedId == R.id.account_login) {
            head_net_url.setVisibility(View.INVISIBLE);
            third_id.setVisibility(View.GONE);
            mTextInputUsername.setVisibility(View.VISIBLE);
            mTextInputPassword.setVisibility(View.VISIBLE);
        } else if (checkedId == R.id.id_login) {
            head_net_url.setVisibility(View.VISIBLE);
            third_id.setVisibility(View.VISIBLE);
            mTextInputUsername.setVisibility(View.VISIBLE);
            mTextInputPassword.setVisibility(View.GONE);
        }
    }
    private ProgressDialog mWaitingDialog;

    public void login(String username, String userPass) {
        VhallSDK.login(username, userPass, new LoginCallbackInternal(userPass));
    }

    private class LoginCallbackInternal implements UserInfoDataSource.UserInfoCallback {

        private String userPass;

        public LoginCallbackInternal() {
        }

        public LoginCallbackInternal(String userPass) {
            this.userPass = userPass;
        }

        @Override
        public void onSuccess(UserInfo userInfo) {
            Toast.makeText(LoginActivity.this, R.string.login_success, Toast.LENGTH_SHORT).show();
            VhallApplication.param.key = userPass;
            skipMain();
            mWaitingDialog.cancel();
        }

        @Override
        public void onError(int errorCode, String reason) {
            Toast.makeText(LoginActivity.this, reason, Toast.LENGTH_SHORT).show();
            mWaitingDialog.cancel();
        }
    }

//    /**
//     * 判断是否缺少权限
//     */
//    private boolean lacksPermission() {
//        return ContextCompat.checkSelfPermission(LoginActivity.this, Manifest.permission.CAMERA) ==
//                PackageManager.PERMISSION_DENIED && ContextCompat.checkSelfPermission(LoginActivity.this, Manifest.permission.RECORD_AUDIO) ==
//                PackageManager.PERMISSION_DENIED;
//    }

    public void loginClick(View view) {
        mWaitingDialog.show();
        if (account_login.isChecked()) {
            checkUserInfo();
        } else {
            loginByThirdId();
        }

    }

    public void registerClick(View view) {
        if (alertDialog == null) {
            AlertDialog.Builder builder = new AlertDialog.Builder(this);
            builder.setMessage(getString(R.string.regist_phone_number, getString(R.string.phone_number)));
            builder.setPositiveButton(R.string.call, new DialogInterface.OnClickListener() {
                @Override
                public void onClick(DialogInterface dialog, int which) {
                    dialog.dismiss();
                    Uri uri = Uri.parse(getString(R.string.phone_number_uri, getString(R.string.phone_number)));
                    Intent intent = new Intent(Intent.ACTION_DIAL, uri);
                    startActivity(intent);
                }
            });
            builder.setNegativeButton(R.string.cancel, new DialogInterface.OnClickListener() {
                @Override
                public void onClick(DialogInterface dialog, int which) {
                    dialog.dismiss();
                }
            });
            alertDialog = builder.create();
        }
        alertDialog.show();
    }

    public void customerClick(View view) {
        skipMain();
    }

    private void skipMain() {
        Intent intent = new Intent(LoginActivity.this, MainActivity.class);
        startActivity(intent);
        finish();
    }

    void loginByThirdId() {
        VhallSDK.loginByThirdId(third_id.getText().toString(), mTextInputUsername.getText().toString(), head_net_url.getText().toString(), new LoginCallbackInternal());
    }


    public void checkUserInfo() {
        String username = mTextInputUsername.getText().toString();
        String password = mTextInputPassword.getText().toString();
        if (!TextUtils.isEmpty(username) && !TextUtils.isEmpty(password)) {
            login(username, password);
        }
    }

    @Override
    public boolean dispatchTouchEvent(MotionEvent ev) {
        return super.dispatchTouchEvent(ev);
    }

}

File: VhallAndroidSDKLiveDemo/uilibs/src/main/java/com/vhall/uimodule/login/LoginActivity.kt
package com.vhall.uimodule.login


import android.content.Intent
import android.text.TextUtils
import android.view.View
import androidx.lifecycle.lifecycleScope
import com.vhall.business.VhallSDK
import com.vhall.business.data.UserInfo
import com.vhall.business.data.source.UserInfoDataSource.UserInfoCallback
import com.vhall.uimodule.BuildConfig
import com.vhall.uimodule.R
import com.vhall.uimodule.base.BaseActivity
import com.vhall.uimodule.dao.UserDataStore
import com.vhall.uimodule.databinding.ActivityLoginBinding
import com.vhall.uimodule.main.MainActivity
import com.vhall.uimodule.utils.CommonUtil
import com.vhall.uimodule.widget.MainListPop
import kotlinx.coroutines.launch

class LoginActivity : BaseActivity<ActivityLoginBinding>(ActivityLoginBinding::inflate) {

    var account = ""
    var passWord = ""
    var id = ""
    var name = ""
    var url = ""

    override fun initView() {
        getData()
        mViewBinding.tvVersion.text = BuildConfig.VH_VERSION_NAME+"("+BuildConfig.BuildTime+")"
        mViewBinding.rgLogin.setOnCheckedChangeListener { group, checkedId ->
            if (checkedId == R.id.account_login) setAccountData()
            else if (checkedId == R.id.id_login) setThirdData()
        }
        mViewBinding.tvJoin.setOnClickListener { loginClick() }
        mViewBinding.tvSign.setOnClickListener { SignConfigActivity.startActivity(mContext) }
        mViewBinding.ivQ.setOnClickListener { showQ() }
        setThirdData()
//        CommonUtil.isGrantedAndRequestTelPermission(this,0)
    }


    private fun loginClick() = if (mViewBinding.accountLogin.isChecked) {
        loginByAccount()
    } else {
        loginByThirdId()
    }

    private fun showQ() = if (mViewBinding.accountLogin.isChecked) {
        MainListPop(mContext, "如何获取三方ID", "三方用户ID是用来关联微吼和外部用户，主播使用的三方用户ID请参考创建直播账号，third_user_id为三方用户ID，pass即为密码").show(mViewBinding.ivQ, 0, 0)
    } else {
        MainListPop(mContext, "如何获取三方ID", "三方用户ID是用来关联微吼和外部用户，您可以选择自己随机设置一串符，点击登录后微吼会自动生成一个对应的用户身份，可以调用OpenAPI接口创建一个三方用户后使用接口中third_user_id来此登录。特别注意头像和昵称只可设置一次且不可修改").show(mViewBinding.ivQ, 0, 0)
    }

    private fun loginByAccount() {
        val text = mViewBinding.edAccount.text.toString()
        val text1 = mViewBinding.edPassword.text.toString()
        if (!TextUtils.isEmpty(text) && !TextUtils.isEmpty(text1)) {
            VhallSDK.loginByAccount(text, text1, LoginCallbackInternal())
        }
    }

    private fun loginByThirdId() {
        if (!TextUtils.isEmpty(mViewBinding.edAccount.text)) {
            VhallSDK.loginByThirdId(
                mViewBinding.edAccount.text.toString(),
                mViewBinding.edPassword.text.toString(),
                mViewBinding.edHead.text.toString(),
                LoginCallbackInternal()
            )
        }
    }

    private fun getData() {
        lifecycleScope.launch {
            account = UserDataStore.getAccount(mContext)
            passWord = UserDataStore.getPassword(mContext)
            id = UserDataStore.getThirdId(mContext)
            name = UserDataStore.getThirdName(mContext)
            url = UserDataStore.getThirdPic(mContext)
            mViewBinding.edAccount.setText(account)
            mViewBinding.edPassword.setText(passWord)
        }
    }

    private fun saveData() {
        lifecycleScope.launch {
            if (mViewBinding.accountLogin.isChecked()) {
                UserDataStore.saveAccount(
                    mViewBinding.edAccount.text.toString(),
                    mViewBinding.edPassword.text.toString(),
                    mContext
                )
            } else {
                UserDataStore.saveThirdId(
                    mViewBinding.edAccount.text.toString(),
                    mViewBinding.edPassword.text.toString(),
                    mViewBinding.edHead.text.toString(),
                    mContext
                )
            }
        }
    }

    private fun setAccountData() {
        id = mViewBinding.edAccount.text.toString()
        name = mViewBinding.edPassword.text.toString()
        url = mViewBinding.edHead.text.toString()

        mViewBinding.edHead.visibility = View.GONE
        mViewBinding.edAccount.setText(account)
        mViewBinding.edPassword.setText(passWord)
        mViewBinding.edAccount.hint = "账号"
        mViewBinding.edPassword.hint = "密码"
    }

    private fun setThirdData() {
        account = mViewBinding.edAccount.text.toString()
        passWord = mViewBinding.edPassword.text.toString()
        mViewBinding.edHead.visibility = View.VISIBLE
        mViewBinding.edAccount.setText(id)
        mViewBinding.edPassword.setText(name)
        mViewBinding.edHead.setText(url)
        mViewBinding.edAccount.hint = "三方用户ID"
        mViewBinding.edPassword.hint = getString(R.string.app_login_nickname)
    }

    inner class LoginCallbackInternal : UserInfoCallback {

        override fun onSuccess(userInfo: UserInfo) {
            saveData()
            startActivity(Intent(mContext, MainActivity::class.java))
            finish()
        }

        override fun onError(errorCode: Int, reason: String) {
            showToast(reason)
        }
    }

}